### PR TITLE
Feature | gds assessment | your details and change your details pages

### PR DESCRIPTION
**Pull request by omarCreativeDev**
- renamed `user-details` component to `your-details` component to be consistent with `your-account` component naming
- renamed `change-user-details` component to `change-your-details` component to be consistent with `your-account` component naming
- made `your-details` a base class and ensured `change-your-details` component extends from it as the pages are effectively almost the same and by doing so removed lots of duplicate code
- general tidy up and refactor

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[your details page](https://trello.com/c/gHJVQLIH/492-your-details-page)

File: src/app/features/registration/create-username/create-username.component.ts [hunk 1]
```diff
@@ -162,7 +162,7 @@ export class CreateUsernameComponent implements OnInit {
 
     this.currentSection = this.currentSection + 1;
 
-    if (this.backLink === '/user-details') {
+    if (this.backLink === '/your-details') {
```
Comment (ChazUK) [inline src/app/features/registration/create-username/create-username.component.ts]: Does this work with the new back link service?

File: src/app/features/registration/create-username/create-username.component.ts [hunk 2]
```diff
@@ -209,16 +209,17 @@ export class CreateUsernameComponent implements OnInit {
...
-      const createPasswordValue = this.registration.locationdata[0].user.password;
-
-      this.createUserNamePasswordForm.setValue({
-        createUsernameInput: createUsernameValue,
-        passwordGroup: {
-          createPasswordInput: createPasswordValue,
-          confirmPasswordInput: createPasswordValue,
-        },
-      });
+      // TODO refactor due to model change
```
Comment (ChazUK) [inline src/app/features/registration/create-username/create-username.component.ts]: Can this be removed if it's all commented out?

File: src/app/features/registration/create-username/create-username.component.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/features/registration/create-username/create-username.component.ts]: im working on the `create-username` page in my next pr so will pick up this change then :)

File: src/app/features/registration/create-username/create-username.component.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/features/registration/create-username/create-username.component.ts]: im working on the `create-username` page in my next pr so will pick up this change then :)

**Pull request merged**